components/org.wso2.carbon.identity.organization.management.organization.user.sharing/src/main/java/org/wso2/carbon/identity/organization/management/organization/user/sharing/listener/SharedUserOperationEventListener.java

```diff
@@ -79,13 +114,226 @@ public boolean doPreDeleteUserWithID(String userID, UserStoreManager userStoreMa
         }
     }
 
+    @Override
+    public boolean doPostGetUserClaimValuesWithID(String userID, String[] claims, String profileName,
+                                                  Map<String, String> claimMap, UserStoreManager userStoreManager)
+            throws UserStoreException {
+
+        if (!isEnable() || userStoreManager == null) {
+            return true;
+        }
+        /*
+        If shared user profile update governance event listener is not enabled, skip shared profile resolver.
+         */
+        if (!sharedUserProfileUpdateGovernanceEventListener.isEnable()) {
+            return true;
+        }
+        String currentTenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        String currentOrganizationId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getOrganizationId();
+        try {
+            if (!OrganizationManagementUtil.isOrganization(currentTenantDomain)) {
+                // There is no shared users in root organizations. Hence, return.
+                return true;
+            }
+            currentOrganizationId = resolveOrganizationId(currentTenantDomain, currentOrganizationId);
```

Comment:
```suggestion
            currentOrganizationId = resolveOrganizationId(currentTenantDomain, currentOrganizationId);
```
change as resolveOrganizationId(currentTenantDomain)

Reply:
fixed in [fa50618](https://github.com/wso2-extensions/identity-organization-management/pull/427/commits/fa506184962f22207ffd0e556949fca70c00d7b4)